Sources/VLCMediaDataSource.m

```diff
@@ -89,11 +89,9 @@ - (void)moveObjectFromIndex:(NSUInteger)fromIdx toIndex:(NSUInteger)toIdx
             return;
         [_foundMedia removeObjectAtIndex:fromIdx];
         [_foundMedia insertObject:object atIndex:toIdx];
-        object.folderTrackNumber = @(toIdx - 1);
-        object = [_foundMedia objectAtIndex:fromIdx];
-        if (![object isKindOfClass:[MLFile class]])
-            return;
-        object.folderTrackNumber = @(fromIdx - 1);
+        for (MLFile *object in _foundMedia) {
+            object.folderTrackNumber = @([_foundMedia indexOfObject:object]);
+        }
```

Comment:
Since `indexOfObject` requires a linear search, this method has probably a complexity of O(n), that together with the wrapping for loop means having a quadratic complexity.

If I understand this code correctly, I think we could improve the performance of this piece of code to be linear just using a counter to get the current index iteration without the need of searching.

```
var count = 0;
for (MLFile *object in _foundMedia) {
  object.folderTrackNumber = @(count);
  count++;
}
```

Reply:
didn't even think of that ![](https://media1.giphy.com/media/26ufdipQqU2lhNA4g/giphy.gif)
thanks @dcordero !